apps/meet-app/backend/modules/database/db_functions.bal

```diff
@@ -1,62 +1,100 @@
+
+    return {
+        domain: meetingTypes.domain,
+        types: types
+    };
+}
+
+# Create new meeting.
 #
-# + name - Name to filter  
-# + 'limit - Limit of the response  
-# + offset - Offset of the number of sample collection to retrieve
-# + return - List of sample collections|Error
-public isolated function fetchSampleCollections(string? name, int? 'limit, int? offset) returns SampleCollection[]|error {
-    stream<SampleCollection, error?> resultStream = databaseClient->
-            query(getSampleCollectionsQuery(name, 'limit, offset));
-
-    SampleCollection[] sampleCollections = [];
-    check from SampleCollection sampleCollection in resultStream
+# + addMeetingPayload - Meeting details
+# + createdBy - Person who created the meeting
+# + return - Id of the meeting | Error
+public isolated function addMeeting(AddMeetingPayload addMeetingPayload, string createdBy) returns int|error {
+    sql:ExecutionResult|error executionResults = databaseClient->execute(
+        addMeetingQuery(addMeetingPayload, createdBy));
+    if executionResults is error {
+        return executionResults;
+    }
+
+    return executionResults.lastInsertId.ensureType(int);
+}
+
+# Fetch meetings.
+#
+# + title - Name to filter  
+# + host - Host email filter  
+# + startTime - Start time filter  
+# + endTime - End time filter  
+# + wso2Participants - WSO2 participants filter
+# + 'limit - Limit of the response
+# + offset - Offset of the number of meetings to retrieve  
+# + return - List of meetings | Error
+public isolated function fetchMeetings(string? title, string? host, string? startTime, string? endTime,
+        string? wso2Participants, int? 'limit, int? offset) returns Meeting[]|error {
+
+    stream<Meeting, error?> resultStream = databaseClient->
+                query(getMeetingsQuery(title, host, startTime, endTime, wso2Participants, 'limit, offset));
+
+    Meeting[] meetings = [];
+
+    check from Meeting meeting in resultStream
         do {
-            sampleCollections.push(sampleCollection);
+            meetings.push(meeting);
         };
 
-    return sampleCollections;
+    return meetings;
 }
 
-# Fetch specific sample collection.
+# Fetch specific meeting.
 #
-# + id - Identification of the sample collection
-# + return - Sample collections|Error, if so
-public isolated function fetchSampleCollection(int id) returns SampleCollection|error? {
-    SampleCollection|sql:Error sampleCollection = databaseClient->queryRow(getSampleCollectionQuery(id));
+# + meetingId - The ID of the meeting to fetch
+# + return - Meeting | Error, if not found
+public isolated function fetchMeeting(int meetingId) returns Meeting|error? {
+    Meeting|sql:Error meeting = databaseClient->queryRow(getMeetingQuery(meetingId));
 
-    if sampleCollection is sql:Error && sampleCollection is sql:NoRowsError {
+    if meeting is sql:Error && meeting is sql:NoRowsError {
         return;
     }
-    return sampleCollection;
+    return meeting;
```

Comment:
```suggestion
    if meeting is sql:Error && meeting is sql:NoRowsError {
        return;
    }
    if meeting is sql:Error {
        return meeting;
    }
    return meeting;
```

Reply:
Since we return same `meeting` object here, do we need the`if meeting is sql:Error` condition?